### PR TITLE
feat: Improve integration tests

### PR DESCRIPTION
**Pull request by Hofer-Julian**
- Test locally build pixi, instead of installed one
- Add task `test-integration-dev` with dependency on task `build`
- Use list instead of strings for commands, so that it also works if the paths contain spaces

### CODE REVIEW

File: .github/workflows/test_downstream.yml [hunk 1]
```diff
@@ -117,8 +117,10 @@ jobs:
     name: Integration test on ${{ inputs.os }}
     runs-on: ${{ inputs.os }}
     steps:
-      - name: checkout repo
+      - name: Checkout repo
         uses: actions/checkout@v4
+      - name: Setup pixi
+        uses: prefix-dev/setup-pixi@v0.8.1
```
Comment (nichmor) [inline .github/workflows/test_downstream.yml:123]: could you please explain why we need to use setup-pixi action here and why it's better than old approach?

I fear that using setup-pixi we allow us to make accidently errors in moving binary and stuff, and still have a working pixi provided by setup-pixi which is not possible using the old approach

Comment (nichmor) [reply .github/workflows/test_downstream.yml:123]: I agree with an env variable for testing purposes

File: pixi.toml [hunk 2]
```diff
@@ -30,7 +32,10 @@ pytest = ">=8.0.2,<9"
 
 [feature.pytest.tasks]
 test-integration = "pytest tests/integration"
```
Comment (nichmor) [inline pixi.toml:34]: can we also pass here PIXI_PATH=None or something to ensure that we do not use by accident pixi setup in a path?

Comment (Hofer-Julian) [reply pixi.toml:34]: I now take `pixi` always from a specific path.
That way there's no option for accidents, I also removed the second task as you requested.

**Review by nichmor — APPROVED** [review]

**Pull request merged**